### DISCUSSION TITLE
Can not install

### DISCUSSION THREAD

**Discussion by ssb6695** [Genuitec/Copilot4Eclipse-community#41] [Help] [answered]
Hi there,

I'm trying to install copilot4elicpse, but it doesn't work.
It seems that my PC can't access "https://www.genuitec.com/updates/copilot4eclipse/content.xml". The response status code is 504.
Is there any trouble on the server side?

**Comment by wayneparrott** [comment]
Are you installing directly from the Eclipse Marketplace via the Eclipse Marketplace Client (MPC) in Eclipse? If yes please try again and share your results. Copilot4Eclipse is downloaded from our CDN that may have had a region glitch.

Wayne
Copilot4Eclipse Team

**Reply by ssb6695** [reply]
Thank you for replying to my inquiry.

> Are you installing directly from the Eclipse Marketplace via the Eclipse Marketplace Client (MPC) in Eclipse?
Yes, I am.
I tried to install it again, but the result was the same as yesterday.

I'll share the screenshot of the error below.
![image](https://github.com/user-attachments/assets/899fd638-d476-469a-ac8c-a44ffe20a183)

I am trying this operation from Japan. If there is a problem with the CDN in a region around Japan, I would appreciate it if you could resolve it.

**Comment by wayneparrott** [comment]
The error msg is really helpful here to see the PKIX/certificate problem.

Please see this technical note for how to enable your JRE running Eclipse to be able to verify the TLS certificate chain to a trusted certificate authority (CA): 
https://www.genuitec.com/products/copilot4eclipse/docs/resources/technotes/installation-pkix-fix

Let us know if the technote helps resolve the install issue.

Wayne
Copilot4Eclipse Team

**Reply by ssb6695** [reply]
Thank you for replying to me and sorry for the late reply.
It seems that I have to talk to the security administrator in my organization. It's predicted that it will take a few weeks.
When I can resolve this phenomenon, I will report it here.

**Reply by toddewilliams** [reply]
This type of issue can happen with companies that tightly control access to external sites. The good news is that there is a corporate option available that allows internal packaging and redistribution of Copilot4Eclipse so it can be used without changes to your external security protocols. You read all about it [here](https://www.genuitec.com/products/copilot4eclipse/docs/legal-info/corporate-license).

**Comment by ssb6695** [comment, marked as answer]
I'm sorry for my late reply.

Finally, I could resolve the problem.
The error was solved after adding 2 lines below at the end of the eclipse.ini file.

-Djava.net.preferIPv4Stack=true
-Djavax.net.ssl.trustStoreType=Windows-ROOT
